src/commands/sourcemaps/inject.rs

```diff
@@ -28,22 +29,28 @@ pub fn make_command(command: Command) -> Command {
             Arg::new("path")
                 .value_name("PATH")
                 .required(true)
-                .help("The path or glob to the javascript files."),
+                .help("The path to the javascript files."),
         )
         .hide(true)
 }
 
 pub fn execute(matches: &ArgMatches) -> Result<()> {
-    let path = matches.get_one::<String>("path").unwrap();
+    let path = matches.get_one::<OsString>("path").unwrap();
 
-    let collected_paths: Vec<PathBuf> = glob(path)
-        .unwrap()
-        .flatten()
-        .filter(|path| path.extension().map_or(false, |ext| ext == "js"))
-        .collect();
+    let mut collected_paths = Vec::new();
```

Comment:
Wanted to see how it can be rewritten, so in case you want to use any of those. Im fine with a simple for loop as well though :P

```rust
let collected_paths: Vec<_> = WalkDir::new(path)
    .into_iter()
    .filter_map(|entry| {
        entry.ok().map(|entry| {
            if entry.path().extension().map_or(false, |ext| ext == "js") {
                Some(entry.path().to_owned())
            } else {
                None
            }
        })
    })
    .flatten()
    .collect();
```

```rust
let collected_paths: Vec<_> = WalkDir::new(path)
    .into_iter()
    .filter_map(|entry| {
        entry
            .ok()
            .map(|entry| {
                if entry.path().extension().map_or(false, |ext| ext == "js") {
                    Some(entry.path().to_owned())
                } else {
                    None
                }
            })
            .flatten()
    })
    .collect();
```

```rust
let collected_paths: Vec<_> = WalkDir::new(path)
    .into_iter()
    .filter_map(Result::ok)
    .flat_map(|entry| {
        entry.path().extension().map(|ext| {
            if ext == "js" {
                Some(entry.path().to_owned())
            } else {
                None
            }
        })
    })
    .flatten()
    .collect();
```

Reply:
I played around for a bit with this as well, and tbh I think all iterator versions are worse than the loop in this case. Also, your versions differ from mine semantically in that they silently discard errors, whereas mine unwraps. Which do you think is preferable in this situation?

Reply:
I'd log and continue. This is what we do when we do a recursive lookup for difs.